opal/runtime/opal_init_core.c

```diff
@@ -407,10 +408,11 @@ int opal_init_gethostname(void)
          * the buffer and try again.
          */
         length *= 2;
-        buf = realloc(buf, length);
+        newbuf = realloc(buf, length);
         if (NULL == buf) {
             return OPAL_ERR_OUT_OF_RESOURCE;
         }
+        buf = newbuf;
```

Comment:
This will still leak `buf` if we return `OPAL_ERR_OUT_OF_RESOURCE`. Maybe something like this is needed?
```suggestion
        newbuf = realloc(buf, length);
        if (NULL == newbuf) {
            free(buf);
            return OPAL_ERR_OUT_OF_RESOURCE;
        }
        buf = newbuf;
```

Reply:
+1 with what @jjhursey said -- I was typing the same comment, and he beat me to it.  😄